DAOS-17629 unified log time stamp format test

grom72 opened the pull request:
### Steps for the author:

* [ ] Commit message follows the [guidelines](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Appropriate [Features or Test-tag](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags) pragmas were used.
* [ ] Appropriate [Functional Test Stages](https://daosio.atlassian.net/wiki/spaces/DC/pages/12147556353/CI+Functional+Test+Stages) were run.
* [ ] At least two positive code reviews including at least one code owner from each category referenced in the PR.
* [ ] Testing is complete. If necessary, forced-landing label added and a reason added in a comment.

#### After all prior steps are complete:
* [ ] Gatekeeper requested (daos-gatekeeper added as a reviewer).

grom72 committed:
Use custom pipeline-lib to avoid pipeline failure

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/5/execution/node/502/log

grom72 committed:
Custom NLT agent

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true

Skip-func-hw-test: true
Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
DAOS-17629 engine: Engine time log format to follow server's one

DAOS engine's log line should include the year and microseconds in
the time stampformat.

DAOS engine log line should exclude '-' as a date and time separator
in favour of space.

Adjust the log parser for the new date and time format:
- calculate the beginning of the message based on the actual field length
- data and time are two separate fields.
  Increase the indexes of remaining fields by one.

Allow-unstable-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage Build on EL 9 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/16/execution/node/314/log

grom72 committed:
Priority at the beggining of the log message

Priority shall be placed at the beginning of the log messages as it is
defined in sys-log standard and to follow the same format as it is
defined in Go loger.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/27/execution/node/502/log

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/28/execution/node/502/log

daosbuild3 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/32/execution/node/1016/log

grom72 committed:
Catch pid conversion exception

Catch errors in line decoding
- too short line
- invalid spliting
- invalid PIDs value

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-tests:true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: true

Allow-unstable-test: truegit

Skip-func-test-el8: false
Skip-fault-injection-test: false

Skip-func-hw-test: true
Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
More precise log line parsing

Since the CART log line is now more similar to the DAOS server log line
(almost the same header), the parsing mechanism must be more robust and
rely not only on field length but also on detailed analysis of the
content.

The reason for this is that the CART log parser is used in some tests
against all logs produced by the DAOS system.
This can lead to test failure due to unrecognized/invalid log format.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Define preamble only once

preamble used in to_str method is calculated only one time based on
explicity fields list including full time tag and facility.

Cancel-prev-build: false

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: false
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: false
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Try to cache log file with failed iterator

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
NLT.9 DEBUG with old message lines detection mechanism

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/39/execution/node/502/log

grom72 committed:
%d may produce "-1" for TID/PID with value 0xFFFFFFFF

Valid type for PID/TID is pid_t.
pid_t may be negative.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/40/execution/node/502/log

daosbuild3 commented:
Test stage Build on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/41/execution/node/317/log

daosbuild3 commented:
Test stage Build on EL 9 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/41/execution/node/318/log

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16789/42/testReport/

grom72 committed:
Exception based approach in _load_data

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Fault injection testing in parallel with NLT tests

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Exception based approach in _load_pids

Cancel-prev-build: false

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Exception based approach in __lnext

Cancel-prev-build: false

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Skip DAOS server message interpretation (from *.go)

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Revert "Exception based approach in __lnext"

This reverts commit 59d895f12b5ae520b2fdfda08ea380ab3714ebb3.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Reapply "Exception based approach in __lnext"

This reverts commit 613def80a76d3f22622ecbfb5d6a844a38912ac5.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Revert "Reapply "Exception based approach in __lnext""

This reverts commit 70905cbca3d06d5c10f2ee50eed08633442081ae.

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: false
Skip-unit-test-memcheck: false
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Revert "Use custom pipeline-lib to avoid pipeline failure"

This reverts commit eff56e8dfe411b3e6de0c4e4b26e149cf99010c9.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
DEBUG (#47): 'NoneType' object has no attribute 'filename'

Cancel-prev-build: false

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
DEBUG: original log levels

Cancel-prev-build: false

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
DEBUG: nlt file not found exception

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
FIX: CaRT file name reference may contain .go

The only way to distinguish between CaRT log line and server log line
(with *.go as file reference) is to check that instead of host name
the Go file reference is detected in 3rd field of the log line.

Cancel-prev-build: false

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Update documentation, solve pylint issues

No need to rerun Functional and HW Tests

Cancel-prev-build: false

Skip-func-test-el8: true

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage Build on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/53/execution/node/318/log

grom72 committed:
Fix: proper pid_t type usage

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
DEBUG: NLT extended diagnostic

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true

Skip-func-hw-test: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Some debugging

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: true

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: true
Skip-funct-test-vm: true

Skip-test: false

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Merge branch 'master' into grom72/DAOS-17629-unified-log-time-stamp-format-test

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Update after merge with master

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-icc: true
Skip-build-ubuntu20-gcc: true

Skip-build-el9-rpm: true
Skip-build-leap15-rpm: true

Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-NLT: false

Allow-unstable-test: true

Skip-func-test-el8: true
Skip-fault-injection-test: false
Skip-funct-test-vm: true

Skip-test: false

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage Functional Hardware Medium MD on SSD completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/65/execution/node/915/log

daosbuild3 commented:
Test stage Functional Hardware Large MD on SSD completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/66/execution/node/1117/log

grom72 committed:
Merge remote-tracking branch 'origin/master' into grom72/DAOS-17629-unified-log-time-stamp-format-test

Stage: Build - Build on EL 9
Stage: Build - Build on Leap 15.5

Skip-unit-test: true
Skip-unit-test-memcheck: true

Allow-unstable-test: true

Skip-func-test-vm-valgrind: true
Skip-func-test-el9: true
Skip-func-test-leap15: true
Skip-test-el-8.6-rpms: true
Skip-test-leap-15-rpms: true

Skip-func-hw-test-medium: true
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

daosbuild3 commented:
Test stage Functional Hardware Medium Verbs Provider MD on SSD completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/68/execution/node/979/log

daosbuild3 commented:
Test stage Functional Hardware Medium MD on SSD completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/68/execution/node/1024/log

grom72 committed:
Merge remote-tracking branch 'origin/master' into grom72/DAOS-17629-unified-log-time-stamp-format-test

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true

Skip-unit-test: true
Skip-unit-test-memcheck: true

    Allow-unstable-test: true

Skip-func-test-vm-valgrind: true
Skip-func-test-el9: true
Skip-func-test-leap15: true
Skip-test-el-8.6-rpms: true
Skip-test-leap-15-rpms: true

Skip-func-hw-test-medium: true
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: true

    Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Merge remote-tracking branch 'origin/master' into grom72/DAOS-17629-unified-log-time-stamp-format-test

Skip-build-el9-gcc: true
Skip-build-leap15-gcc: true

Skip-unit-test: true
Skip-unit-test-memcheck: true

Allow-unstable-test: true

Skip-func-test-vm-valgrind: true
Skip-func-test-el9: true
Skip-func-test-leap15: true
Skip-test-el-8.6-rpms: true
Skip-test-leap-15-rpms: true

Skip-func-hw-test-medium: true
Skip-func-hw-test-medium-verbs-provider-md-on-ssd: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

grom72 committed:
Trigger build

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

Allow-unstable-test: true

daosbuild3 commented:
Test stage Build on Leap 15.5 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/73/execution/node/405/log

daosbuild3 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/74/execution/node/262/log

daosbuild3 commented:
Test stage Build on EL 9.6 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/74/execution/node/337/log

grom72 committed:
Update raft

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>

Allow-unstable-test: true

daosbuild3 commented:
Test stage Functional Hardware Medium MD on SSD completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16789/75/execution/node/1340/log
